### PR TITLE
User Permissions and Date fix

### LINKED ISSUE #45: Logged in user cant register
If a user is currently logged in they still see the form to register

### LINKED ISSUE #46: Editing a Viewing Party Issue
When editing a viewing party, leaving the date/time blank throws an error.

```ruby
ActionView::Template::Error (undefined method `map' for nil:NilClass):

    25:   <span>Include: <br/>
    26:     <%= collection_check_boxes :friend, :ids, @invitees, :id, :name, include_hidden: false %>
    27:   </span>
    28:   <%= submit_tag "Update Party" %>
    29: </section>
  
app/views/viewing_party/edit.html.erb:26:in `block in _app_views_viewing_party_edit_html_erb___1197658802995943691_70313661158080'
app/views/viewing_party/edit.html.erb:6:in `_app_views_viewing_party_edit_html_erb___1197658802995943691_70313661158080'
app/controllers/viewing_party_controller.rb:46:in `update'
```

### PR DESCRIPTION
**Pull request by ETBassist**
- Create test for registration page redirect
- Add redirect for current user in users#new
- Add expectations for require_login for viewing party pages
- Add require_login to ApplicationController and ViewingPartyController
- Add pre-filled values to edit party date and time
- Add sad path test for editing Party
- Add default date and time values to edit form
- Add test for @invitees conditional in Party#edit view
- Add minimum date to input on party#edit & #new

X____ Wrote Tests X____ Implemented ____ Reviewed

Neccesary checkmarks:

    [x] All Tests are Passing

    [] The code will run locally

Type of change

    [] New feature
    [x] Bug Fix

Implements/Fixes:

Closes #45
Closes #46

Check the correct boxes

    [x] This broke nothing
    [] This broke some stuff
    [] This broke everything

Testing Changes

    [x] No Tests have been changed
    [] Some Tests have been changed
    [] All of the Tests have been changed(Please describe what in the world happened)

Checklist:

    [x] My code has no unused/commented out code
    [] I have reviewed my code
    [] I have commented my code, particularly in hard-to-understand areas
    [x] I have fully tested my code

(For Fun!)Please Include a link to a gif of your feelings about this branch

Link:

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -10,4 +10,9 @@ def current_user
   def require_user
     render file: 'public/404', status: :not_found unless current_user
   end
+
+  def require_login
+    flash[:notice] = 'Please log in or register to visit this page' unless current_user
+    redirect_to '/' unless current_user
+  end
```
Comment (helloeduardo) [inline app/controllers/application_controller.rb:17]: This is great to redirect to the home page!

File: app/views/viewing_party/edit.html.erb [hunk 2]
```diff
@@ -9,11 +9,11 @@
 
   Currently selected date for your Party: <%=@party.date%> <br>
   <label for="date">Select a new date for your viewing party:</label> <br>
-  <input type="date" id="date" name="date"><br/><br>
+  <input type="date" id="date" name="date" min="<%= Date.today %>" value="<%= @party.date %>"><br/><br>
```
Comment (helloeduardo) [inline app/views/viewing_party/edit.html.erb:12]: I like that this uses the party's existing date!

**Review by helloeduardo — APPROVED** [review]
Great additions overall for some quality of life improvements

File: app/views/viewing_party/edit.html.erb (hunk 2 above)
Comment (Jonathan-M-Wilson) [inline app/views/viewing_party/edit.html.erb:12]: Wow! This is great! Thank you for fixing this!

**Pull request merged**